### DISCUSSION TITLE
How to switch to different tex engines for different projects?

### DISCUSSION THREAD

**Discussion by xlucn** [latex-lsp/texlab#972] [Q&A] [answered]
If I have two projects, where I want to build one with xelatex, but the other with pdflatex. How do I make texlab choose the correct options for latexmk? Should I change the lsp settings, or should I put some configuration files inside the projects?

Currently I have these in neovim's config
```
... ...
build = {
                executable = "latexmk",
                args = { "-outdir=./output/", "-xelatex", "-synctex=1",
                         "-interaction=nonstopmode", "%f" },
            }
... ...
```
where texlab will always use xelatex for all projects.

**Comment by pfoerster** [comment, marked as answer]
@xlucn Some clients allow having project-local settings (e. g. `.vscode/settings.json`). A more universal solution would be to set the engine in the `latexmkrc` file (use the `$pdflatex` variable) and omit the `-xelatex` argument in the `texlab` config.

**Reply by xlucn** [reply]
The latexmkrc solution seems promising. Thanks, I will try and come back later!

**Reply by NiklasKappel** [reply]
I have the same problem, but I don't normally use latexmkrc files. What would be a minimal `latexmkrc`, that behaves the same as the default texlab config, and only sets the executable to xelatex?

**Reply by xlucn** [reply]
> I have the same problem, but I don't normally use latexmkrc files. What would be a minimal `latexmkrc`, that behaves the same as the default texlab config, and only sets the executable to xelatex?

@NiklasKappel I think adding this to your `latexmkrc` will do
```
$pdf_mode = 5;
```

Or use `$pdflatex` option, as suggested by @pfoerster. I found the documents here: https://texdoc.org/serve/latexmk/0
